### ISSUE TITLE
Two anonymous functions passed as function/method arguments cause indentation false positive

### ISSUE THREAD

**Issue by Ingramz** [squizlabs/PHP_CodeSniffer#961]
Sample code

```php
<?php

namespace MyNamespace;

class SomeClass
{
    public function someFunc()
    {
        a(function () {
            echo "a";
        })->b(function () {
            echo "b";
        });

        if (true) {
            echo "c";
        }
        echo "d";
    }
}
```

Running `phpcs --standard=psr2 .` results in
```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 15 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8
 17 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found
    |       |     8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

Note the presence of the `if` statement (can also be something like `for`). If one omits this part from the code, it will pass successfully. Outdenting the `if` block will work as well in order to pass the test, but I don't think this what we want.

**Comment by gsherwood** [comment]
Debug output showing the problem (just for my records):
```
$ phpcs temp.php --standard=Generic --sniffs=Generic.WhiteSpace.ScopeIndent --runtime-set scope_indent_debug 1
Start with token 0 on line 1 with indent 0
Open scope (T_CLASS) on line 6
	=> indent set to 4 by token 12 (T_OPEN_CURLY_BRACKET)
Open scope (T_FUNCTION) on line 8
	=> indent set to 8 by token 24 (T_OPEN_CURLY_BRACKET)
Open closure on line 9
	=> indent set to 12 by token 27 (T_STRING)
Close scope (T_CLOSURE) on line 11
	=> indent set to 8 by token 43 (T_CLOSE_CURLY_BRACKET)
Close closure on line 11
	* token has nested parenthesis 28 on line 9 *
	* token is inside condition 17 (T_FUNCTION) on line 7 *
	* using parenthesis *
	* previous token is T_STRING on line 9 *
	* first token on line 9 is 27 (T_STRING) *
	=> indent set to 8 by token 27 (T_STRING)
Open closure on line 11
	=> indent set to 12 by token 43 (T_CLOSE_CURLY_BRACKET)
Close scope (T_CLOSURE) on line 13
	=> indent set to 8 by token 62 (T_CLOSE_CURLY_BRACKET)
Close closure on line 13
	* token has nested parenthesis 47 on line 11 *
	* token is inside condition 17 (T_FUNCTION) on line 7 *
	* using parenthesis *
	* previous token is T_STRING on line 11 *
	* first token on line 11 is 43 (T_CLOSE_CURLY_BRACKET) *
	* first token is a scope closer *
	=> indent set to 4 by token 43 (T_CLOSE_CURLY_BRACKET)
[Line 15] Line indented incorrectly; expected 4 spaces, found 8
	=> Add adjustment of -4 for token 68 (T_IF) on line 15
Open scope (T_IF) on line 15
	=> indent set to 8 by token 74 (T_OPEN_CURLY_BRACKET)
Indent adjusted to 8 for T_ECHO on line 16
	=> Add adjustment of -4 for token 77 (T_ECHO) on line 16
Close scope (T_IF) on line 17
	=> indent set to 4 by token 83 (T_CLOSE_CURLY_BRACKET)
[Line 17] Line indented incorrectly; expected 4 spaces, found 8
	=> Add adjustment of -4 for token 83 (T_CLOSE_CURLY_BRACKET) on line 17
Close scope (T_FUNCTION) on line 19
	=> indent set to 4 by token 92 (T_CLOSE_CURLY_BRACKET)
Close scope (T_CLASS) on line 20
	=> indent set to 0 by token 94 (T_CLOSE_CURLY_BRACKET)
```

**Comment by samhsuqld** [comment]
I can confirm this issue: it works in 2.5.0 but not 2.5.1. The first commit that causes the code to error is 103e2f0

**Comment by gsherwood** [comment]
Should be fixed now. Tried with a couple of other code samples from #1002 and they are working as well.

**Comment by hskrasek** [comment]
Thanks @gsherwood
